File: javascript/hexcompress_spec.js
require('should');

var hexcompress = require('../lib/hexcompress');

describe('hexcompress.decompress', function() {

  it('keeps ascii', function() {
    hexcompress.decompress(new Buffer("asd")).should.equal("asd");
  });

  it('decodes hex', function() {
    hexcompress.decompress(new Buffer([0x61, 0x73, 0xdf])).should.equal("asdf");
  });

  it('decodes capital ascii', function() {
    hexcompress.decompress(new Buffer("asdF")).should.equal("asdF");
  });

  it('decodes single hex', function() {
    hexcompress.decompress(new Buffer([0x1f])).should.equal("1f");
  });

  it('decodes single hex with follow-up', function() {
    hexcompress.decompress(new Buffer([0x1f, 0x41])).should.equal("1fA");
  });

  it('decodes single hex with prefix', function() {
    hexcompress.decompress(new Buffer([0x7a, 0x1f])).should.equal("z1f");
  });
  
  it('keeps 00', function() {
    hexcompress.decompress(new Buffer("00")).should.equal("00");
  });

  it('keeps null byte', function() {
    hexcompress.decompress(new Buffer([0x00])).should.equal("\u0000");
  });

  it('keeps 41', function() {
    hexcompress.decompress(new Buffer("41")).should.equal("41");
  });

  it('unpacks trailing zeroes', function() {
    hexcompress.decompress(new Buffer([0x01, 0x02])).should.equal("0102");
  });
});

File: ruby/spec/hexcompress_spec.rb
require_relative '../hexcompress.rb'

describe 'Hexcompress.decompress' do

  it 'keeps ascii' do
    Hexcompress.decompress("asd").should eq("asd")
  end

  it 'decodes hex' do
    Hexcompress.decompress("\x61\x73\xdf".encode("ASCII-8BIT")).should eq("asdf")
  end

  it 'decodes capital ascii' do
    Hexcompress.decompress("asdF").should eq("asdF")
  end

  it 'decodes single hex' do
    Hexcompress.decompress("\x1f".encode("ASCII-8BIT")).should eq("1f")
  end

  it 'decodes single hex with follow-up' do
    Hexcompress.decompress("\x1f\x41".encode("ASCII-8BIT")).should eq("1fA")
  end

  it 'decodes single hex with prefix' do
    Hexcompress.decompress("\x7a\x1f".encode("ASCII-8BIT")).should eq("z1f")
  end
  
  it 'keeps 00' do
    Hexcompress.decompress("00").should eq("00")
  end

  it 'keeps null byte' do
    Hexcompress.decompress("\x00".encode("ASCII-8BIT")).should eq("\u0000")
  end

  it 'keeps 41' do
    Hexcompress.decompress("41").should eq("41")
  end

  it 'unpacks trailing zeroes' do
    Hexcompress.decompress("\x01\x02".encode("ASCII-8BIT")).should eq("0102")
  end

end
